### PR TITLE
Let empty queries export everything

### PR DESCRIPTION
**Pull request by dominiklohmann**
### :notebook_with_decorative_cover: Description

This changes the empty query, i.e., either a non-provided query argument or an empty string as the query argument, to export everything. Under the hood this is implemented by transforming empty queries into a query that parses everything, namely `#type != "this expression matches everything"`.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Try it out.

### CODE REVIEW

**Review by mavam — COMMENTED** [review]
Why do you modify the expression parser for this? If the calling context provides the empty string, you can still fill in the somewhat arbitrary expression.

**Comment by dominiklohmann** [comment]
> Why do you modify the expression parser for this? If the calling context provides the empty string, you can still fill in the somewhat arbitrary expression.

Because that requires touching four pieces of code instead of one, which is much less maintainable.

File: libvast/src/system/read_query.cpp [hunk 1]
```diff
@@ -60,21 +60,17 @@ read_query(const invocation& inv, std::string_view file_option,
...
     }
-  } else if (inv.arguments.empty()) {
-    // Read query from STDIN.
-    if (::isatty(::fileno(stdout)))
-      std::cerr << "please enter a query and confirm with CTRL-D: "
-                << std::flush;
-    assign_query(std::cin);
+  } else if (inv.arguments.size() <= argument_offset) {
+    VAST_VERBOSE("not providing a query causes everything to be exported; "
+                 "please be aware that this operation may be very expensive.");
```
Comment (lava) [inline libvast/src/system/read_query.cpp:65]: I guess it doesn't hurt, but when testing the only way I was able to see this message was when I was providing a limit, ie. when I was already working around the thing it warns against :D

**Review by lava — APPROVED** [review]
Looks simple enough, and I've tested locally that it works as expected.

It does a bit more than what was discussed in the story; we may want to reconsider if we want to restrict this feature to `export` since for expose and pivot it is a bit unintuitive, and for `vast count` a user may expect a more performant implementation for the operation of counting the number of items in the database.

File: doc/cli/vast-pivot.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ type exists in the schema, VAST will dynamically create a new query to fetch the
 contextual data according to the type relationship.
 
 ```bash
-vast pivot [options] <type> <expr>
+vast pivot [options] <type> [<expr>]
 ```
 
 VAST uses the field `community_id` to pivot between logs and packets. Pivoting
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply doc/cli/vast-pivot.md:10]: Their docs do not contain the command line interface at all. I will add it.

File: libvast/src/system/read_query.cpp [hunk 3]
```diff
@@ -60,21 +60,17 @@ read_query(const invocation& inv, std::string_view file_option,
                                "unable to read from " + *fname);
       assign_query(f);
     }
-  } else if (inv.arguments.empty()) {
-    // Read query from STDIN.
-    if (::isatty(::fileno(stdout)))
-      std::cerr << "please enter a query and confirm with CTRL-D: "
-                << std::flush;
-    assign_query(std::cin);
+  } else if (inv.arguments.size() <= argument_offset) {
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/src/system/read_query.cpp:63]: That's a good point. I will make that change and disable it for explore and pivot.

File: libvast/src/system/read_query.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/system/read_query.cpp:65]: You can still see it in the log file and also you can see it if you're piping the results somewhere (or writing to a file directly).

**Pull request merged**